From: Ryusuke Konishi <konishi.ryusuke@gmail.com>
Subject: nilfs2: use integer type instead of enum req_op for event tracing header
Date: Tue, 30 Apr 2024 17:00:18 +0900

Patch series "nilfs2: reduce build warnings with "make C=1""

These two are cleanups that eliminate the warnings output by sparse (not
including common one related to refcount_dec_and_lock).  Neither is a
bugfix.  


This patch (of 2):

The sparse check with "make C=1" outputs warnings regarding references to
the header file "include/trace/events/nilfs2.h" for event tracing:

 fs/nilfs2/segment.c: note: in included file (through
   include/trace/trace_events.h, include/trace/define_trace.h,
   include/trace/events/nilfs2.h):
 ./include/trace/events/nilfs2.h:191:1: warning: cast to restricted
   blk_opf_t
 ./include/trace/events/nilfs2.h:191:1: warning: restricted blk_opf_t
   degrades to integer
 ./include/trace/events/nilfs2.h:191:1: warning: restricted blk_opf_t
   degrades to integer

Fix this issue by reverting the type of the parameter related to the bio
operation mode in the event tracing definition from "enum req_op" to
"int".

In order to prevent sparse warnings on the caller side (where
trace_nilfs2_mdt_submit_block() is used), also add a typecast to an
argument passed to the tracepoint.

Link: https://lkml.kernel.org/r/20240430080019.4242-1-konishi.ryusuke@gmail.com
Link: https://lkml.kernel.org/r/20240430080019.4242-2-konishi.ryusuke@gmail.com
Signed-off-by: Ryusuke Konishi <konishi.ryusuke@gmail.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202401092241.I4mm9OWl-lkp@intel.com/
Fixes: ed4512590bd5 ("fs/nilfs2: Use the enum req_op and blk_opf_t types")
Cc: Bart Van Assche <bvanassche@acm.org>
Cc: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/nilfs2/mdt.c               |    2 +-
 include/trace/events/nilfs2.h |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/fs/nilfs2/mdt.c~nilfs2-use-integer-type-instead-of-enum-req_op-for-event-tracing-header
+++ a/fs/nilfs2/mdt.c
@@ -152,7 +152,7 @@ nilfs_mdt_submit_block(struct inode *ino
 	ret = 0;
 
 	trace_nilfs2_mdt_submit_block(inode, inode->i_ino, blkoff,
-				      opf & REQ_OP_MASK);
+				      (__force int)(opf & REQ_OP_MASK));
  out:
 	get_bh(bh);
 	*out_bh = bh;
--- a/include/trace/events/nilfs2.h~nilfs2-use-integer-type-instead-of-enum-req_op-for-event-tracing-header
+++ a/include/trace/events/nilfs2.h
@@ -192,7 +192,7 @@ TRACE_EVENT(nilfs2_mdt_submit_block,
 	    TP_PROTO(struct inode *inode,
 		     unsigned long ino,
 		     unsigned long blkoff,
-		     enum req_op mode),
+		     int mode),
 
 	    TP_ARGS(inode, ino, blkoff, mode),
 
@@ -200,7 +200,7 @@ TRACE_EVENT(nilfs2_mdt_submit_block,
 		    __field(struct inode *, inode)
 		    __field(unsigned long, ino)
 		    __field(unsigned long, blkoff)
-		    __field(enum req_op, mode)
+		    __field(int, mode)
 	    ),
 
 	    TP_fast_assign(
_
